# Define the goal

#"zoom", "zoom_point", "focus", "EV", "ISO", "aperture" , "shutter_speed"
string mode

#for zoom
float32 zoom_factor

#for zoom_point
uint8 zoom_point_multiplier #Data struct of tap zoom multiplier used by user. Limit: 1~5
float32 zoom_point_x  # range from 0~1, top left corner is origin point
float32 zoom_point_y  # range from 0~1, top left corner is origin point

#for focus
float32 focus_x #range from 0~1, top left corner is origin point
float32 focus_y #range from 0~1, top left corner is origin point

#for "EV", "ISO", "aperture" , "shutter_speed"
uint8 exposure_mode #see enum class ExposureMode in common_type.h

uint8 exposure_compensation #see enum class ExposureCompensation in common_type.h
uint16 aperture      #see enum class Aperture in common_type.h
uint8 iso_data #see enum class ISO in common_type.h
uint8 shutter_speed #see enum class ShutterSpeed in common_type.h

---
# Define the result
bool done
---
# Define a feedback message
float32 rate